editoast/editoast_derive/src/model.rs

```diff
@@ -1,55 +1,231 @@
-use darling::FromDeriveInput;
+use darling::{FromDeriveInput, Result};
+use proc_macro2::TokenStream;
 use quote::quote;
-use syn::{parse_macro_input, DeriveInput};
+use syn::{DeriveInput, Ident};
+
+use crate::case::to_snake_case;
 
 #[derive(FromDeriveInput)]
-#[darling(attributes(model))]
+#[darling(attributes(model), forward_attrs(allow, doc, cfg))]
 struct ModelOptions {
+    /// Table of the model
+    table: syn::Path,
+    /// Enable the `create` functions
+    create: Option<bool>,
+    /// Enable the `load` functions
+    retrieve: Option<bool>,
+    /// Enable the `delete` functions
+    delete: Option<bool>,
+}
+
+struct Config {
```

Comment:
It may be useful to also allow the giving the macro a NotFound variant of an existing Error enum instead of always creating a new one. Because afaik there is no way to add more variants to an existing enum, so let's say I want to have an Error type related to Documents with more custom errors, I would need to create `DocumentErrorBis` to hold the additional variants and end up with two error types related to Document.

Reply:
Giving a variant is not convenient. It will have to match a specific pattern (it must be a named variant with only one field with a specific name). In addition, the pattern to have generic errors created automatically by the macro and specific ones created by the user if needed is not that ugly.